tools/caffe.cpp

```diff
@@ -12,13 +12,18 @@ using caffe::Blob;
 using caffe::Caffe;
 using caffe::Net;
 using caffe::Layer;
+using caffe::Solver;
 using caffe::shared_ptr;
+using caffe::string;
 using caffe::Timer;
 using caffe::vector;
-
+using std::ostringstream;
 
 DEFINE_int32(gpu, -1,
-    "Run in GPU mode on given device ID.");
+    "Run in GPU mode on given device ID (Legacy switch, use -gpus).");
```

Comment:
Having both `gpu` and `gpus` seems unecessary. Why introduce a new arg instead of redefining `gpu` as a string and including the logic to handle a single device number, a comma separated list, or "all" ?